### PR TITLE
Siblings = overlapping events

### PR DESCRIPTION
**Pull request by bkniffler**
Change the sibbling detection and recognize all overlapping events

### CODE REVIEW

File: src/utils/dayViewLayout.js [hunk 1]
```diff
@@ -34,20 +34,27 @@ let getSlot = (event, accessor, min, totalMin) => event && positionFromDate(
...
  */
-let isSibling = (idx1, idx2, { events, startAccessor, min, totalMin }) => {
-  let event1 = events[idx1]
-  let event2 = events[idx2]
+var isSibling = function isSibling(idx1, idx2, _ref2) {
+  var events = _ref2.events,
+      startAccessor = _ref2.startAccessor,
+      endAccessor = _ref2.endAccessor,
+      min = _ref2.min,
+      totalMin = _ref2.totalMin;
```
Comment (jquense) [inline src/utils/dayViewLayout.js]: this looks like babel output...

**Comment by bkniffler** [comment]
Urgh, sorry bout the code, I've been working on it through my node_modules to make sure it works, obviously been using the compiled source. Should be okay now.

**Comment by bkniffler** [comment]
Just as a comparison (with 1px white border on bottom):

Before:
xyz

After:
xy

It would be supercool if this could be merged soon, its one of the things currently holding me up to give access to testers.

**Comment by tobiasandersen** [comment]
I agree that the "after" screenshot you posted looks better, but this PR is gonna affect more complex layouts quite a lot. See e.g. the stories I created when implementing this:

**Set 0**

Before:
0-current

After:
0-pr

**Set 1**

Before:
1-current

After:
1-pr

I'd prefer if we could keep those layouts as is (since you can fit much more), but adjust for the situation where the end and start is the same (so that we get your after layout). Something like this should do just that: 

```
let isSibling = (idx1, idx2, { events, startAccessor, endAccessor, min, totalMin }) => {
  let event1 = events[idx1]
  let event2 = events[idx2]

  if (!event1 || !event2) return false

  let start1 = getSlot(event1, startAccessor, min, totalMin)
  let start2 = getSlot(event2, startAccessor, min, totalMin)
  let end1 = getSlot(event1, endAccessor, min, totalMin)

  return (Math.abs(start1 - start2) < 60 && start2 < end1)
}

```

That `60` number is still quite arbitrary, but will be fixed with https://github.com/intljusticemission/react-big-calendar/pull/340/commits/ef0017975f53e03c79937fed2ed854a95cdea9f8.

File: src/utils/dayViewLayout.js [hunk 2]
```diff
@@ -34,19 +34,20 @@ let getSlot = (event, accessor, min, totalMin) => event && positionFromDate(
...
-let isSibling = (idx1, idx2, { events, startAccessor, min, totalMin }) => {
+let isSibling = (idx1, idx2, { events, startAccessor, endAccessor, min, totalMin }) => {
   let event1 = events[idx1]
   let event2 = events[idx2]
 
   if (!event1 || !event2) return false
 
-  let start1 = getSlot(event1, startAccessor, min, totalMin)
-  let start2 = getSlot(event2, startAccessor, min, totalMin)
+  var start1 = getSlot(event1, startAccessor, min, totalMin)
```
Comment (tobiasandersen) [inline src/utils/dayViewLayout.js]: Small knit pick, but can we keep `let`?

**Comment by jquense** [comment]
Ha I missed all that when I looked. Thanks for double checking :)

**Comment by bkniffler** [comment]
Hey @tobiasandersen, your code works fine for my requirements, so I updated my PR. The `var`s shouldn't be an issue anymore.

**Comment by AgamlaRage** [comment]
@bkniffler I tried your PR (utils/dayViewLayout.js) modifications, works fine for me. 
Can confirm that fixed the issue #378 for me.

I have an other behavior that is unexpected with event overlaping, I post it here because I think it's related with this PR. Here the save events.

**GoogleCalendar/Fullcalendar**

rbc-issue-fullcalendar-xs

**React Big Calendar**

rbc-issue-react-big-calendar-xs

As I say in the image, **Blue event should be bellow the red one, and stop at the same position that the yellow one**.

My question is:

Is that plan to have the exact same behavior as GoogleCalendar/Fullcalendar ?
If yes, Is that related to this PR ?
If not, should I open a new issue ?

Thx

**Comment by bkniffler** [comment]
It seems like big calendar tries to follow google calendar in terms of look&feel, which is good since the google devs surely put lots of research into their calendar. I bet the maintainers here would be willing to accept a PR on this matter.

There is 2 things that could to be addressed somewhere around [here](https://github.com/bkniffler/react-big-calendar/blob/4ed1442f9ca69227d16e22c91c475c2076054cba/src/utils/dayViewLayout.js#L196) to imitate gcal:
- z-index of sibling events needs to be raised (through inline styles or via an additional class `rdc-sibling-event`) or rendering of these events needs to happen after non-sibling ones.
- only the left offset should vary with `siblingIdx` rising, while still trying to fill the whole width. Currently the algorithm will make the yellow 70% width/0 offset. It could be better to follow google calendar and make the yellow 100% width/0 offset, the red 70% width/30% offset. 

Give it a shot, shouldn't be too hard!

**Comment by AgamlaRage** [comment]
Ok thx ! I'll do that
I open a new issue #404  to ge the tracker clean.

**Pull request merged**